polaris-core/src/main/java/org/apache/polaris/core/config/PolarisConfiguration.java

```diff
@@ -86,18 +102,32 @@ public static List<PolarisConfiguration<?>> getAllConfigurations() {
   @SuppressWarnings("unchecked")
   protected PolarisConfiguration(
       String key,
+      Set<String> legacyKeys,
       String description,
       T defaultValue,
       Optional<String> catalogConfig,
       Optional<String> catalogConfigUnsafe) {
     this.key = key;
+    this.legacyKeys = legacyKeys;
     this.description = description;
     this.defaultValue = defaultValue;
     this.catalogConfigImpl = catalogConfig;
     this.catalogConfigUnsafeImpl = catalogConfigUnsafe;
     this.typ = (Class<T>) defaultValue.getClass();
   }
 
+  public boolean hasLegacyKeys() {
+    return !legacyKeys.isEmpty();
+  }
+
+  public String legacyKeys() {
+    if (legacyKeys.isEmpty()) {
+      throw new IllegalStateException(
+          "Attempted to read legacy keys from a configuration that doesn't have any.");
```

Comment:
This doesn't look right to me; the check is there for `catalogConfig` because it's not correct to call the variant of `getConfiguration` which takes a catalog when a config doesn't have a `catalogConfig`. However we should always be checking legacyKeys every time we load any config.

Reply:
I've removed it.